src/core/infra/repositories/tasks.repository.ts

```diff
@@ -5,24 +5,66 @@ import { mapTaskEntityFromDbModel } from '../mappers/task-entity-from-db-model-m
 
 const RESOURCE_NAME = 'Task';
 
-async function findTasksByProjectId (id_project: string) : Promise<Task[]> {
+/**
+ * Creates a new task in the database.
+ *
+ * @param new_task: Task - New task to be created.
+ * @returns {Promise<Task>} - Created task.
+ *
+ * @throws {Error} - If an error occurs when creating the task.
+ */
+async function createTask(newTask: Task): Promise<Task | null> {
+  return await Prisma.$transaction(async (prisma: any) => {
     try {
-        let data = await Prisma.task.findMany({
-            where: {
-                id_project: id_project
-            },
-        });
-    
-        if (!data) {
-            throw new NotFoundError(RESOURCE_NAME);
-        }
-    
-        return data.map(mapTaskEntityFromDbModel);
-        
-    } catch (error: unknown) {
-        throw new Error(`${RESOURCE_NAME} repository error`);
+      const existingTask = await prisma.task.findUnique({
+        where: { id: newTask.id },
+      });
+
+      if (existingTask) {
+        console.log(`Task with id ${newTask.id} already exists`);
+        return null;
+      }
```

Comment:
Try using the KISS method (Keep it simple and stupid). If you find the need of making another method, use it.

Reply:
Es una transaction, no le veo necesidad de separar el find

Reply:
Necesitamos tener consistencia en el codigo

Reply:
ya lo dividí el find con el create, pero el data va a ser todo